ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -587,40 +682,54 @@ def _useRealization(self, info, rlz):
                                      self.rank,
                                      self.crowdingDistance,
                                      self.objectiveVal,
-                                     self.constraints,
+                                     self.fitness,
                                      self.constraintsV)
           self._resolveNewGenerationMulti(traj, rlz, info)
 
           ##############################################################################
-          objs_vals = [list(ele) for ele in list(zip(*self.objectiveVal))]
           ##TODO: remove all the plots and maybe design new plots in outstreams if our current cannot be used
           ## These are currently for debugging purposes
           import matplotlib.pyplot as plt
-          # JY: Visualization: all points - This code block needs to be either deleted or revisited.
-          plt.plot(np.array(objs_vals)[:,0], np.array(objs_vals)[:,1],'*')
-
-          # JY: Visualization: optimal points only - This code block needs to be either deleted or revisited.
-          # plt.xlim(75,100)
-          # plt.ylim(5,20)
-          # plt.xlim(0,1)
-          # plt.ylim(0,6)
-          plt.title(str('Iteration ' + str(self.counter-1)))
-
-          plt.plot(np.array(list(zip(self._optPointHistory[traj][-1][0][self._objectiveVar[0]], self._optPointHistory[traj][-1][0][self._objectiveVar[1]])))[:,0],
-                   np.array(list(zip(self._optPointHistory[traj][-1][0][self._objectiveVar[0]], self._optPointHistory[traj][-1][0][self._objectiveVar[1]])))[:,1],'*')
-          for i in range(len(np.array(list(zip(self._optPointHistory[traj][-1][0][self._objectiveVar[0]], self._optPointHistory[traj][-1][0][self._objectiveVar[1]])))[:,0])):
-            plt.text(np.array(list(zip(self._optPointHistory[traj][-1][0][self._objectiveVar[0]], self._optPointHistory[traj][-1][0][self._objectiveVar[1]])))[i,0],
-                     np.array(list(zip(self._optPointHistory[traj][-1][0][self._objectiveVar[0]], self._optPointHistory[traj][-1][0][self._objectiveVar[1]])))[i,1], str(self.batchId-1))
-            plt.savefig('PF'+str(i)+'_'+str(self.counter-1)+'.png')
+
+          signChange = list(map(lambda x:-1 if x=="max" else 1 , self._minMax))
+          for i in range(0, len(self.multiBestObjective)):
+            newMultiBestObjective = self.multiBestObjective * signChange
+
```

Comment:
This line (696) doesn't change in the loop (is not a function of i). What is the significance of the loop? I am sure, I am missing something.

Reply:
This plotting lines will be revisited to make animations rather than a series of snapshots later and temporarily commented.